### PR TITLE
samba: fix darwin build

### PR DESCRIPTION
**Pull request by jacereda**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by strager** [comment]
It looks like samba is still broken on macOS (10.12) after this PR:

```
$ nix-build -A samba .
/nix/store/1jaskd4fvvhdp9i9bxjkhvvz1n5jkfmy-samba-4.10.2

$ /nix/store/1jaskd4fvvhdp9i9bxjkhvvz1n5jkfmy-samba-4.10.2/bin/smbstatus
dyld: Library not loaded: /private/tmp/nix-build-samba-4.10.2.drv-0/samba-4.10.2/bin/default/librpc/libndr.dylib
  Referenced from: /nix/store/1jaskd4fvvhdp9i9bxjkhvvz1n5jkfmy-samba-4.10.2/bin/smbstatus
  Reason: image not found
zsh: abort      /nix/store/1jaskd4fvvhdp9i9bxjkhvvz1n5jkfmy-samba-4.10.2/bin/smbstatus

$ /nix/store/1jaskd4fvvhdp9i9bxjkhvvz1n5jkfmy-samba-4.10.2/bin/smbd
dyld: Library not loaded: /private/tmp/nix-build-samba-4.10.2.drv-0/samba-4.10.2/bin/default/lib/util/libsamba-util.dylib
  Referenced from: /nix/store/1jaskd4fvvhdp9i9bxjkhvvz1n5jkfmy-samba-4.10.2/bin/smbd
  Reason: image not found
zsh: abort      /nix/store/1jaskd4fvvhdp9i9bxjkhvvz1n5jkfmy-samba-4.10.2/bin/smbd
```

**Comment by jacereda** [comment]
Waiting for an answer: https://github.com/NixOS/nixpkgs/issues/59925

**Comment by jacereda** [comment]
In any case I would merge this, since it's preventing things like `ffmpeg-full` from building.

File: pkgs/servers/samba/4.x.nix [hunk 1]
```diff
@@ -70,6 +75,7 @@ stdenv.mkDerivation rec {
       "--enable-fhs"
       "--sysconfdir=/etc"
       "--localstatedir=/var"
+      "--disable-rpath"
```
Comment (flokli) [inline pkgs/servers/samba/4.x.nix]: should this be behind a `optional stdenv.isDarwin`?

Comment (jacereda) [reply pkgs/servers/samba/4.x.nix]: Not sure. According to https://wiki.samba.org/index.php/Waf 

```
For distribution packagers, rpath is rarely wanted, and is often forbidden. 
So if you are making a distribution of Samba or one of our standalone libraries, 
you should use:

waf configure --disable-rpath
```

Comment (flokli) [reply pkgs/servers/samba/4.x.nix]: I did build both variants, the binaries basically look the same:

```
--- /tmp/foo1	2019-05-22 15:27:28.882083224 +0200
+++ /tmp/foo2	2019-05-22 15:27:33.268065308 +0200
@@ -93,7 +93,7 @@
  0x0000000000000001 (NEEDED)             Shared library: [libjansson.so.4]
  0x0000000000000001 (NEEDED)             Shared library: [libz.so.1]
  0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
- 0x000000000000001d (RUNPATH)            Library runpath: [/nix/store/jhgwg5h49bg42vdva80yad5j27mlp0na-samba-4.10.2/lib/samba:/nix/store/xw6qh505pmwsn3q4w5pp488jad18yjlc-krb5-1.17/lib:/nix/store/jhgwg5h49bg42vdva80yad5j27mlp0na-samba-4.10.2/lib:/nix/store/bfq0l29ifpn9r3yi6viz9dvmc7s528gf-popt-1.16/lib:/nix/store/j2450pgy4lhp41im86cfzr7mwlcnfx4z-jansson-2.12/lib:/nix/store/iiymx8j7nlar3gc23lfkcscvr61fng8s-zlib-1.2.11/lib:/nix/store/cksgpzwcdywfaqyj4w17w3qp1lszbw2m-libbsd-0.9.1/lib:/nix/store/p5synzdykdch5cgw2qhzqbc4715lzsch-systemd-239.20190219-lib/lib:/nix/store/681354n3k44r8z90m35hm8945vsp95h1-glibc-2.27/lib]
+ 0x000000000000001d (RUNPATH)            Library runpath: [/nix/store/69mi99d2c6pgdls45j0zxpsrwah2di24-samba-4.10.2/lib/samba:/nix/store/69mi99d2c6pgdls45j0zxpsrwah2di24-samba-4.10.2/lib:/nix/store/xw6qh505pmwsn3q4w5pp488jad18yjlc-krb5-1.17/lib:/nix/store/bfq0l29ifpn9r3yi6viz9dvmc7s528gf-popt-1.16/lib:/nix/store/j2450pgy4lhp41im86cfzr7mwlcnfx4z-jansson-2.12/lib:/nix/store/iiymx8j7nlar3gc23lfkcscvr61fng8s-zlib-1.2.11/lib:/nix/store/cksgpzwcdywfaqyj4w17w3qp1lszbw2m-libbsd-0.9.1/lib:/nix/store/p5synzdykdch5cgw2qhzqbc4715lzsch-systemd-239.20190219-lib/lib:/nix/store/681354n3k44r8z90m35hm8945vsp95h1-glibc-2.27/lib]
  0x000000000000000c (INIT)               0x2000
  0x000000000000000d (FINI)               0x2d44
  0x0000000000000019 (INIT_ARRAY)         0x46b0
```

So I guess it's fine to keep stick to what upstream recommends for packaging.

**Pull request merged**